### ISSUE TITLE
Change the "Master" nomenclature

### ISSUE THREAD

**Issue by tlfeng** [opensearch-project/sql#463]
**Is your feature request related to a problem? Please describe.**
[OpenSearch](https://github.com/opensearch-project/OpenSearch) repository is going to replace the terminology "master"with "cluster manager".
issue: https://github.com/opensearch-project/OpenSearch/issues/472, with the plan for its terminology replacement.

Although the existing usages with "master" will be supported in OpenSearch version 2.x to keep the backwards compatibility, please prepare for the nomenclature change in advance, and replace all the usages with "master" terminology in the code base.
All the OpenSearch REST APIs and settings that contain "master" terminology will be deprecated in 2.0, and alternative usages will be added.

**Describe the solution you'd like**
Replace the terminology "master" with "cluster manager".

When being compatible with OpenSearch 2.0:
- Deprecate REST API (request parameter) that contains "master" and provide alternative inclusive usage:
deprecate parameter "master_timeout" and add new parameter "cluster_manager_timeout" in [RestQuerySettingsAction](https://github.com/opensearch-project/sql/blob/1.3.1.0/plugin/src/main/java/org/opensearch/sql/plugin/rest/RestQuerySettingsAction.java#L83) (updated after the [below comment](https://github.com/opensearch-project/sql/issues/463#issuecomment-1103212154))

When being compatible with OpenSearch 3.0:
- Replace "master" in the reference to OpenSearch Java API
  - [x] `clusterUpdateSettingsRequest.masterNodeTimeout` -> `clusterUpdateSettingsRequest.clusterManagerNodeTimeout` (new API is available in version 2.2)

**Describe alternatives you've considered**
None.

**Additional context**

**Comment by dblock** [comment]
I am still seeing https://github.com/opensearch-project/sql/blob/907019cc8f7ce05c34ed453a75e4d897f55b57c5/plugin/src/main/java/org/opensearch/sql/plugin/rest/RestQuerySettingsAction.java#L82

**Comment by dai-chen** [comment]
> I am still seeing
> 
> https://github.com/opensearch-project/sql/blob/907019cc8f7ce05c34ed453a75e4d897f55b57c5/plugin/src/main/java/org/opensearch/sql/plugin/rest/RestQuerySettingsAction.java#L82
> 
> reopening

I didn't see the new API for this. Do we have a doc for this?

**Comment by tlfeng** [comment]
@dai-chen Thanks for taking care of the issue.
The API request parameter "master_timeout" is deprecated and will be replaced by parameter "cluster_manager_timeout".
I requested the documentation update here: https://github.com/opensearch-project/documentation-website/issues/450.
You could take it as a reference.

**Comment by dai-chen** [comment]
> 

@tlfeng Thanks for the very detailed info! Will refresh OpenSearch dependency and check the latest API. Thanks

**Comment by tlfeng** [comment]
To clarify, there are 2 usages with "master" word in this line of code: 
```
clusterUpdateSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterUpdateSettingsRequest.masterNodeTimeout()));
```
1. `ClusterUpdateSettingsRequest.masterNodeTimeout()`
It's an OpenSearch Java API, and need to wait for the OpenSearch (core) to make the name change before version 3.0. 
It can only changed when being compatible with OpenSearch 3.0 .
2. `"master_timeout"`
It's a REST API request parameter. In this use case, the parameter is not only involved in the REST API of SQL plugin, but also transmits to OpenSearch REST API call. 
It's recommended to use `cluster_manager_timeout` instead in version 2.0, but please consider the backwards compatibility. 
It has to be changed when being compatible with OpenSearch 3.0, otherwise deprecation warning and log with be generated due to the use of `master_timeout` in OpenSearch REST API call.

There is no impact for not making changes on the 2  "master" usages in OpenSearch 2.0, so I'm fine to close this issue.
But I recommend to deprecate the use of `"master_timeout"` properly in version 2.x .

**Comment by dai-chen** [comment]
As @tlfeng clarified above, closing this for now. Feel free to comment and reopen if concern.

**Comment by dblock** [comment]
I think this shouldn't have been closed, we're just saying "deprecate usage of master_timeout in 2.1", I'll reopen.

**Comment by dai-chen** [comment]
> I think this shouldn't have been closed, we're just saying "stop using master_timeout in 2.1", I'll reopen.
> 
> https://github.com/opensearch-project/sql/blob/907019cc8f7ce05c34ed453a75e4d897f55b57c5/plugin/src/main/java/org/opensearch/sql/plugin/rest/RestQuerySettingsAction.java#L82

Sure. Once OpenSearch API change complete, this can be fixed accordingly.

**Comment by anasalkouz** [comment]
OpenSearch Java APIs are completed and released on 2.2.
@dai-chen @anirudha  Could you please proceed and complete the remaining part from your end. Let's target to finish and close this on 2.3.

**Comment by dai-chen** [comment]
> OpenSearch Java APIs are completed and released on 2.2. @dai-chen @anirudha Could you please proceed and complete the remaining part from your end. Let's target to finish and close this on 2.3.

Sure, will work on this.
